src/main/java/com/google/sps/api/GmailAPI.java

```diff
@@ -0,0 +1,142 @@
+	public static Gmail getGmailService() throws IOException, GeneralSecurityException {
+
+    // Gather client info from OAuth 2.0 credentials.
+		InputStream in = new FileInputStream(CREDENTIALS_FILE_PATH); 
+		GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+    CLIENT_ID = clientSecrets.getDetails().getClientId().toString();
+    CLIENT_SECRET = clientSecrets.getDetails().getClientSecret().toString();
+
+    // Set up credentials to use the Gmail API.
+		Credential authorize = new GoogleCredential.Builder().setTransport(GoogleNetHttpTransport.newTrustedTransport())
+				.setJsonFactory(JSON_FACTORY)
+				.setClientSecrets(CLIENT_ID, CLIENT_SECRET)
+				.build()
+        .setRefreshToken(Token.REFRESH_TOKEN)
+        .setAccessToken(getAccessToken(Token.REFRESH_TOKEN));
+
+		// Build service.
+		final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+		service = new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, authorize)
+				.setApplicationName(APPLICATION_NAME).build();
+
+		return service;
+	}
+
+  /**
+    * Retrieve new access token due to the fact that it has a lifetime of 1 hour
+    * through refresh token.
+    *
+    * @param refresh_token Authorized token to create access tokens.
+    */
+	private static String getAccessToken(String refresh_token) {
+
+		try {
+
+      // Gather POST parameters.
+			Map<String, Object> params = new LinkedHashMap<>();
+			params.put("grant_type", "refresh_token");
+			params.put("client_id", CLIENT_ID);
+			params.put("client_secret", CLIENT_SECRET); 
+			params.put("refresh_token", refresh_token);
+
+      // Build POST request.
+			StringBuilder postData = new StringBuilder();
+			for (Map.Entry<String, Object> param : params.entrySet()) {
+				if (postData.length() != 0) {
+					postData.append('&');
+				}
+				postData.append(URLEncoder.encode(param.getKey(), "UTF-8"));
+				postData.append('=');
+				postData.append(URLEncoder.encode(String.valueOf(param.getValue()), "UTF-8"));
+			}
+
+			byte[] postDataBytes = postData.toString().getBytes("UTF-8");
+
+      // Send POST request for new access token.
+			URL url = new URL("https://accounts.google.com/o/oauth2/token");
+			HttpURLConnection con = (HttpURLConnection) url.openConnection();
+			con.setDoOutput(true);
+			con.setUseCaches(false);
+			con.setRequestMethod("POST");
+			con.getOutputStream().write(postDataBytes);
+
+      // Read POST response.
+			BufferedReader reader = new BufferedReader(new InputStreamReader(con.getInputStream()));
+			StringBuffer buffer = new StringBuffer();
+			for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+				buffer.append(line);
+			}
+
+      // Return select access token from POST response.
+			JSONObject json = new JSONObject(buffer.toString());
+			String accessToken = json.getString("access_token");
+			return accessToken;
+		
+    } catch (Exception ex) {
+			ex.printStackTrace();
```

Comment:
Explain why you only print the stacktrace e.g. why it's okay to ignore the error.

Reply:
If the POST request to create a new access token does not work, it will include a response with an error description so exceptions for the request wouldn't appear. Additionally, I just configured the tokens so there shouldn't be any current exceptions related to the request. In the future, I will add more exception handling!